### PR TITLE
Pub 2054 - Add SJP delta press list

### PR DESCRIPTION
**Pull request by NatashaAlker**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-2054

### Change description ###

Add SJP delta press list (new cases)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/PublicationManagementService.java [hunk 1]
```diff
@@ -165,6 +166,7 @@ public Map<FileType, byte[]> getStoredPublications(UUID artefactId, String userI
             publicationFilesMap.put(PDF, azureBlobService.getBlobFile(artefactId + ".pdf"));
 
             if (SJP_PUBLIC_LIST.equals(artefact.getListType())
+                || SJP_DELTA_PRESS_LIST.equals(artefact.getListType())
                 || SJP_PRESS_LIST.equals(artefact.getListType())) {
                 publicationFilesMap.put(EXCEL, azureBlobService.getBlobFile(artefactId + ".xlsx"));
             } else {
```
Comment (KianKwa) [inline src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/PublicationManagementService.java:172]: The SJP press list language resource files need to be updated with the new title for the PDF. A new language resource file for the SJP press delta list should also be added to override the title.

File: src/test/java/uk/gov/hmcts/reform/pip/channel/management/services/artefactsummary/SjpPressListSummaryConverterTest.java [hunk 2]
```diff
@@ -13,13 +14,14 @@
...
+    void testSjpPressListSummary(ListType listType) throws IOException {
         String[] outputLines;
         try (InputStream mockFile = Thread.currentThread().getContextClassLoader()
             .getResourceAsStream("mocks/sjpPressList.json")) {
             JsonNode payload = new ObjectMapper().readTree(new String(mockFile.readAllBytes()));
-            String result = new ListConversionFactory().getArtefactSummaryConverter(ListType.SJP_PRESS_LIST)
+            String result = new ListConversionFactory().getArtefactSummaryConverter(listType)
                 .convert(payload);
             outputLines = result.split(System.lineSeparator());
         }
```
Comment (KianKwa) [inline src/test/java/uk/gov/hmcts/reform/pip/channel/management/services/artefactsummary/SjpPressListSummaryConverterTest.java:27]: Need to add a new test for SjpPressListFileConverter too

File: src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/PublicationManagementService.java (hunk 1 above)
Comment (KianKwa) [reply src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/PublicationManagementService.java:172]: This has been done

File: src/test/java/uk/gov/hmcts/reform/pip/channel/management/services/artefactsummary/SjpPressListSummaryConverterTest.java (hunk 2 above)
Comment (KianKwa) [reply src/test/java/uk/gov/hmcts/reform/pip/channel/management/services/artefactsummary/SjpPressListSummaryConverterTest.java:27]: New test added

File: src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/filegeneration/SjpPressListFileConverter.java [hunk 3]
```diff
@@ -73,11 +76,14 @@ public String convert(JsonNode jsonBody, Map<String, String> metadata, Map<Strin
      * @return The converted Excel spreadsheet as a byte array.
```
Comment (ChrisS1512) [inline src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/filegeneration/SjpPressListFileConverter.java:76]: list type needs adding to the java doc above

(Similar comment for other changes)

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/filegeneration/SjpPressListFileConverter.java (hunk 3 above)
Comment (KianKwa) [reply src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/filegeneration/SjpPressListFileConverter.java:76]: Now updated

Comment (KianKwa) [reply src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/filegeneration/SjpPressListFileConverter.java:77]: This time the javadoc is updated with the new param for SJP public and SJP press file converter

**Review by ChrisS1512 — APPROVED** [review]

**Review by junaidiqbalmoj — APPROVED** [review]

**Pull request merged**